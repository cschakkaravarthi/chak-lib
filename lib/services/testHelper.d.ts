export declare const shallowRender: (toRender: any) => Record<string, any>;
//# sourceMappingURL=testHelper.d.ts.map