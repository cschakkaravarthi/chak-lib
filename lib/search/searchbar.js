"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var react_bootstrap_1 = require("react-bootstrap");
require("../../scss/_forms.scss");
var images_1 = require("../assets/images");
exports.SearchBar = function (props) {
    return (react_1["default"].createElement(react_bootstrap_1.Form, { onSubmit: props.handleSubmit, className: 'search-form' },
        react_1["default"].createElement(react_bootstrap_1.InputGroup, null,
            react_1["default"].createElement(react_bootstrap_1.InputGroup.Prepend, null,
                react_1["default"].createElement(react_bootstrap_1.Button, { variant: "outline-secondary", type: "submit" },
                    react_1["default"].createElement("img", { src: images_1.images.searchIconWhite, width: 15, height: 15 }))),
            react_1["default"].createElement(react_bootstrap_1.Form.Control, { placeholder: "Search", "aria-label": "Search", value: props.searchTerm, onChange: props.handleChange }))));
};
//# sourceMappingURL=searchbar.js.map