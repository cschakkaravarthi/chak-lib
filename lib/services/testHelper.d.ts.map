{"version":3,"file":"testHelper.d.ts","sourceRoot":"","sources":["../../src/services/testHelper.tsx"],"names":[],"mappings":"AAKA,eAAO,MAAM,aAAa,wCAIzB,CAAC"}