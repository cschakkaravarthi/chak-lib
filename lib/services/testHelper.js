"use strict";
/* eslint @typescript-eslint/no-explicit-any: "off" */
// In this one file we do actually want to use the "any" type. This is a very generic function.
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var shallow_1 = __importDefault(require("react-test-renderer/shallow"));
exports.shallowRender = function (toRender) {
    var renderer = shallow_1["default"].createRenderer();
    renderer.render(toRender);
    return renderer.getRenderOutput();
};
//# sourceMappingURL=testHelper.js.map