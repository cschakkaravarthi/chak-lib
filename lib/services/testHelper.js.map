{"version":3,"file":"testHelper.js","sourceRoot":"","sources":["../../src/services/testHelper.tsx"],"names":[],"mappings":";AAAA,sDAAsD;AACtD,+FAA+F;;;;;AAE/F,wEAA0D;AAE7C,QAAA,aAAa,GAAG,UAAC,QAAa;IACzC,IAAM,QAAQ,GAAG,oBAAe,CAAC,cAAc,EAAE,CAAC;IAClD,QAAQ,CAAC,MAAM,CAAC,QAAQ,CAAC,CAAC;IAC1B,OAAO,QAAQ,CAAC,eAAe,EAAE,CAAC;AACpC,CAAC,CAAC"}