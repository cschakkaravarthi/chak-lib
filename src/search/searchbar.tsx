import React, { FC } from 'react';
import { Form as SourceForm, Button as SourceButton, InputGroup as SourceInputGroup } from 'react-bootstrap';
import '../../scss/_forms.scss';
import { images } from '../assets/images';

export interface SearchBarProps {
  handleSubmit: (e: React.SyntheticEvent) => void;
  handleChange: (e: React.SyntheticEvent) => void;
  searchTerm?: string;
}

export const SearchBar: FC<SearchBarProps> = props => {
  return (
    <SourceForm onSubmit={props.handleSubmit} className="search-form">
      <SourceInputGroup>
        <SourceInputGroup.Prepend>
          <SourceButton variant="outline-secondary" type="submit">
            <img src={images.searchIconWhite} width={15} height={15} />
          </SourceButton>
        </SourceInputGroup.Prepend>
        <SourceForm.Control
          placeholder="Search"
          aria-label="Search"
          value={props.searchTerm}
          onChange={props.handleChange}
        />
      </SourceInputGroup>
    </SourceForm>
  );
};
