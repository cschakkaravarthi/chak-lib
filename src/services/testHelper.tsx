/* eslint @typescript-eslint/no-explicit-any: "off" */
// In this one file we do actually want to use the "any" type. This is a very generic function.

import ShallowRenderer from 'react-test-renderer/shallow';

export const shallowRender = (toRender: any): Record<string, any> => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(toRender);
  return renderer.getRenderOutput();
};
