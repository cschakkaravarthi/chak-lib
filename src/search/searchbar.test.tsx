import React from 'react';

import { shallowRender } from '../services/testHelper';
import { SearchBar } from './searchbar';

describe('SearchBar', () => {
  it('should render correctly', () => {
    const tree = shallowRender(
      <SearchBar
        handleSubmit={() => { alert('clicked'); }}
        handleChange={() => { alert('clicked'); }}
        searchTerm=""
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
