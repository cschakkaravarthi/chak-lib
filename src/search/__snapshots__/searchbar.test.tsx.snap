// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `
<ForwardRef
  className="search-form"
  inline={false}
  onSubmit={[Function]}
>
  <ForwardRef>
    <ForwardRef>
      <ForwardRef
        active={false}
        disabled={false}
        type="submit"
        variant="outline-secondary"
      >
        <img
          height={15}
          src="test-file-stub"
          width={15}
        />
      </ForwardRef>
    </ForwardRef>
    <ForwardRef
      aria-label="Search"
      onChange={[Function]}
      placeholder="Search"
      value=""
    />
  </ForwardRef>
</ForwardRef>
`;
